cdap-ui/app/hydrator/templates/partial/node-config-modal/preview-tab.html

```diff
@@ -14,153 +14,13 @@
-      </h4>
-      <h4 class="message"
-          ng-if="!HydratorPlusPlusNodeConfigCtrl.previewStatus">
-        Input records are not available for stage "{{ HydratorPlusPlusNodeConfigCtrl.state.node.plugin.label }}". Please try running preview again.
-      </h4>
-    </div>
-
-  </div>
-
-  <!-- OUTPUT RECORDS -->
-  <div ng-class="{'col-xs-12': HydratorPlusPlusNodeConfigCtrl.state.isSource, 'preview-records output': true, 'col-xs-6': HydratorPlusPlusNodeConfigCtrl.state.isTransform}"
-      ng-if="!HydratorPlusPlusNodeConfigCtrl.state.isSink && !HydratorPlusPlusNodeConfigCtrl.state.isCondition">
-
-    <h4>Output Records</h4>
-
-    <div ng-repeat="(key, value) in HydratorPlusPlusNodeConfigCtrl.previewData.output">
-
-      <h5 class="stage-title"
-          ng-if="HydratorPlusPlusNodeConfigCtrl.previewData.numOutputStages > 1">
-        {{ key }}
-      </h5>
-      <div ng-if="value.records.length > 0">
-        <!-- OUTPUT RECORDS -->
-        <table class="table table-bordered">
-          <thead>
-            <th ng-repeat="field in ::value.schemaFields">
-              {{ ::field }}
-            </th>
-          </thead>
-
-          <tbody>
-            <tr ng-repeat="row in ::value.records">
-              <td ng-repeat="field in ::value.schemaFields">
-                {{ ::row[field] }}
-              </td>
-            </tr>
-          </tbody>
-        </table>
-      </div>
-
-      <div class="text-center"
-           ng-if="!value.records || value.records.length === 0">
-        <h4 class="message"
-            ng-if="HydratorPlusPlusNodeConfigCtrl.previewStatus &&
-            HydratorPlusPlusNodeConfigCtrl.previewStatus === 'RUNNING'">
-          Output records have not been generated for stage "{{ HydratorPlusPlusNodeConfigCtrl.state.node.plugin.label }}" yet. Please try again in some time.
-        </h4>
-        <h4 class="message"
-            ng-if="HydratorPlusPlusNodeConfigCtrl.previewStatus &&
-            HydratorPlusPlusNodeConfigCtrl.previewStatus !== 'RUNNING'">
-          Output records have not been generated for stage "{{ HydratorPlusPlusNodeConfigCtrl.state.node.plugin.label }}". Please verify your logic, or try sending more data
-        </h4>
-        <h4 class="message"
-            ng-if="!HydratorPlusPlusNodeConfigCtrl.previewStatus">
-          Output records are not available for stage "{{ HydratorPlusPlusNodeConfigCtrl.state.node.plugin.label }}". Please try running preview again.
-        </h4>
-      </div>
-    </div>
-  </div>
-
-  <!-- NEITHER (CONDITION STAGES) -->
-  <div class="col-xs-12 preview-records"
-        ng-if="HydratorPlusPlusNodeConfigCtrl.state.isCondition">
-    <h4>Input Records & Output Records</h4>
-    <div class="text-center">
-      <h4 class="message">
-        Preview data is not supported for condition stages.
-      </h4>
-    </div>
-  </div>
+<div>
+  <preview-data-view 
+  preview-id="HydratorPlusPlusNodeConfigCtrl.previewId"
```

Comment:
Should these fields (`preview-id`, `selected-node-name`, ...) get indented?

Reply:
done!